app/models/seller.rb

```diff
@@ -75,26 +75,25 @@ def amount_raised
 
   # calculates the amount raised from gift cards
   def gift_card_amount
-    GiftCardDetail.joins(:item)
-                  .where(items: {
-                           seller_id: id,
-                           refunded: false
-                         })
-                  .inject(0) do |sum, gift_card|
-      sum + gift_card.amount
-    end
+    GiftCardDetail
+      .joins(:item)
+      .where(items: {
+               seller_id: id,
+               refunded: false
+             })
+      .joins("join (#{GiftCardAmount.latest_amounts_sql}) as la on la.gift_card_detail_id = gift_card_details.id")
```

Comment:
I didn't know that you could do this

Reply:
I just learned it too ahaha, seemed like the best way to do this!